I thought i got display working but realized my error while writing the pull request comment

nxnode opened the pull request:
i did pytest and i tested it in ipython and it's true that it IS passing the tests, BUT the assignment is:

> display() will return a unicode string representing the list as if it were a Python tuple literal: “(12, ‘sam’, 37, ‘tango’)”

and now that I have typed this and tried a few other things I realize my test is wrong. it should testing to assert ll.display() == "(3, "three", "nine")"

back to the drawing board. I know how to undo a merged pull request but not sure about undoing a pull request that hasn't been merged.

nxnode committed:
Diplay is working but I am not entirely convinved it's 100% correct

nxnode committed:
testing print is close but still not there

W-Ely commented on code_fellows/linked_list.py:
```diff
@@ -4,12 +4,21 @@
+                self._length += 1
+
+    def __len__(self):
+        return self.size()
```
`__len__` and `size` should both `return self._length`

nxnode committed:
Assert passing a tuple to LinkedList works as well

W-Ely commented on tests/code_fellows/test_linked_list.py:
```diff
@@ -1,17 +1,20 @@
 # https://codefellows.github.io/sea-python-401d6/assignments/linked_list.html
 
 import pytest
-
+import sys
 from code_fellows.linked_list import LinkedList
+from io import StringIO
```
Nice find on the test for print!
these imports should be arranged per pep008 https://peps.python.org/pep-0008/#imports

> Imports should be grouped in the following order:
> 1. Standard library imports.
> 2. Related third party imports.
> 3. Local application/library specific imports.

> You should put a blank line between each group of imports.

so in this case:
```python
import sys
from io import StringIO

import pytest

from code_fellows.linked_list import LinkedList
```

W-Ely replied:
might see if VS code has the plugin `isort` https://pypi.org/project/isort/

nxnode committed:
removed failing print test - will start new branch for print function/testing

nxnode commented:
There is a built in isort extension but I didn't have it set to automatically do it's thing. I changed it to sort on save.

My print test is failing because it's returning something like this:
`'<code_fellows.linked_list.LinkedList object at 0x7f438122d160>\n'`

So I removed all of the print stuff and will start a new branch for testing and merge this branch.

W-Ely commented:
> So I removed all of the print stuff and will start a new branch for testing and merge this branch.

This exercise has followed an unexpected code/git flow.  The entirety of these kinds of exercises should be in a single reviewed and approved PR.  As if a new complete feature is being added.

So really instead of merging this, continuing to work on it until it is complete, and then merging it would be better.

If you want a new branch, a okay, branch off this branch, then merge in your changes to this branch once they are ready.

nxnode committed:
got print workinggit pull --rebase I forgot it was __repl__ not __print__

nxnode committed:
got print test working with __repr__ finally
